src/Compilers/CSharp/Test/Emit/CodeGen/PatternTests.cs

```diff
@@ -935,6 +935,786 @@ .maxstack  1
+  IL_0004:  callvirt   ""string Person.Name.get""
+  IL_0009:  stloc.0
+  IL_000a:  ldloc.0
+  IL_000b:  brfalse.s  IL_003f
+  IL_000d:  ldloc.0
+  IL_000e:  ldstr      ""Bill""
+  IL_0013:  call       ""bool string.op_Equality(string, string)""
+  IL_0018:  brtrue.s   IL_0029
+  IL_001a:  ldloc.0
+  IL_001b:  ldstr      ""Bob""
+  IL_0020:  call       ""bool string.op_Equality(string, string)""
+  IL_0025:  brtrue.s   IL_0034
+  IL_0027:  br.s       IL_003f
+  IL_0029:  ldstr      ""Hey Bill!""
+  IL_002e:  call       ""void System.Console.WriteLine(string)""
+  IL_0033:  ret
+  IL_0034:  ldstr      ""Hey Bob!""
+  IL_0039:  call       ""void System.Console.WriteLine(string)""
+  IL_003e:  ret
+  IL_003f:  ldstr      ""Hello ""
+  IL_0044:  ldloc.0
+  IL_0045:  ldstr      ""!""
+  IL_004a:  call       ""string string.Concat(string, string, string)""
+  IL_004f:  call       ""void System.Console.WriteLine(string)""
+  IL_0054:  ret
+}");
+        }
+
+        [Fact, WorkItem(34933, "https://github.com/dotnet/roslyn/issues/34933")]
+        public void NoRedundantPropertyRead02()
+        {
+            var source = @"using System;
+
+public class Person {
+    public string Name { get; set; }
+    public int Age;
+}
+
+public class C {
+    public void M(Person p) {
+        switch (p) {
+            case Person { Name: var name, Age: 0}:
+                Console.WriteLine($""Hello baby { name }!"");
+            break;
+            case Person { Name: var name }:
+                Console.WriteLine($""Hello { name }!"");
+            break;
+        }
+    }
+}";
+            var compilation = CreateCompilation(source, options: TestOptions.ReleaseDll);
+            compilation.VerifyDiagnostics();
+            var compVerifier = CompileAndVerify(compilation);
+            compVerifier.VerifyIL("C.M",
+@"{
+  // Code size       64 (0x40)
+  .maxstack  3
+  .locals init (string V_0, //name
+                string V_1) //name
+  IL_0000:  ldarg.1
+  IL_0001:  brfalse.s  IL_003f
+  IL_0003:  ldarg.1
+  IL_0004:  callvirt   ""string Person.Name.get""
+  IL_0009:  stloc.0
+  IL_000a:  ldarg.1
+  IL_000b:  ldfld      ""int Person.Age""
+  IL_0010:  brtrue.s   IL_0028
+  IL_0012:  ldstr      ""Hello baby ""
+  IL_0017:  ldloc.0
+  IL_0018:  ldstr      ""!""
+  IL_001d:  call       ""string string.Concat(string, string, string)""
+  IL_0022:  call       ""void System.Console.WriteLine(string)""
+  IL_0027:  ret
+  IL_0028:  ldloc.0
+  IL_0029:  stloc.1
+  IL_002a:  ldstr      ""Hello ""
+  IL_002f:  ldloc.1
+  IL_0030:  ldstr      ""!""
+  IL_0035:  call       ""string string.Concat(string, string, string)""
+  IL_003a:  call       ""void System.Console.WriteLine(string)""
+  IL_003f:  ret
+}");
+        }
+
+        [Fact, WorkItem(34933, "https://github.com/dotnet/roslyn/issues/34933")]
+        public void NoRedundantPropertyRead03()
+        {
+            var source = @"using System;
+
+public class Person {
+    public string Name { get; set; }
```

Comment:
>public string Name { get; set; } [](start = 4, length = 32)

Consider testing with `abstract` or `virtual` property with no override in derived type.